build.wake

```diff
@@ -103,10 +103,52 @@ def all variant =
 
     Pass (flatten x)
 
+target findCargoPath Unit =
+    require Some cargo_home = getenv "CARGO_HOME"
+    else
+        require Some path = getenv "PATH"
+        else "cargo"
+
+        whichIn path "cargo"
+
+    "{cargo_home}/bin/cargo"
+
+# Builds a cargo project, the Cargo.toml file must be listed as well as the outputs of intrest.
+# All source files must be listed as well. Cargo will not reuse work because the sandbox
+# will not contain its extra info, inside of target.
+# TODO: Add an environment variable to opt-in to localRunner cargo builds if we detect
+#       existing outputs in {pathToCrate}/target/release
+def buildWithCargo (sources: List Path) (pathToCrate: String) (outputs: List String) =
+    require Some path = getenv "PATH"
+    else failWithError "PATH was not set; cargo will not be able to find rustc"
+
+    require Pass outputs =
+        makeExecPlan (findCargoPath Unit, "build", "--release",) sources
+        | setPlanLabel "cargo: {pathToCrate}"
+        | setPlanDirectory pathToCrate
+        | setPlanFnOutputs (\_ map ("{pathToCrate}/target/release/{_}") outputs)
+        | setPlanEnvironment ("PATH={path}",)
```

Comment:
We set path to a constant (which is good because setting it to an env-var would introduce non-determinism)

```
# A topic used to globally add directories to the default PATH of a Plans
export topic path: String

# Default search path for new Plans
export def path: String =
    catWith ":" (subscribe path | distinctBy scmp)

# Add PATH to the default environment
publish environment =
    "PATH={path}", "TERM=xterm-256color", Nil
```

Reply:
Rereading, I think this is going to give the job whatever the exterior-to-wake PATH is

Could we instead inject only the bit to find cargo?
```suggestion
        | prependPlanPath (dirname (findCargoPath Unit))
```

Reply:
Yeah that would be more hygienic. I'm still working on getting all the docker stuff situated unfortunately.